src/SpacetimeDBClient.cs

```diff
@@ -205,6 +203,97 @@ static DbValue Decode(IRemoteTableHandle table, EncodedValue value, out object?
         private static readonly Status Committed = new Status.Committed(default);
         private static readonly Status OutOfEnergy = new Status.OutOfEnergy(default);
 
+        private static ServerMessage DecompressDecodeMessage(byte[] bytes)
+        {
+            using var stream = new MemoryStream(bytes, 1, bytes.Length - 1);
+
+            switch (bytes[0])
+            {
+                // Uncompressed.
+                case 0:
+                    {
+                        using var binaryReader = new BinaryReader(stream);
+                        return new ServerMessage.BSATN().Read(binaryReader);
+                    }
+                // Brotli compressed.
+                case 1:
+                    {
+                        using var decompressedStream = new BrotliStream(stream, CompressionMode.Decompress);
+                        using var binaryReader = new BinaryReader(decompressedStream);
+                        return new ServerMessage.BSATN().Read(binaryReader);
+                    }
+                default:
+                    throw new InvalidOperationException();
```

Comment:
Can we have a reason argument to the exception? "Unknown compression type" or similar

Reply:
Same for the other exceptions